rpcs3/Emu/Cell/Modules/cellSearch.cpp

```diff
@@ -42,31 +44,93 @@ void fmt_class_string<CellSearchError>::format(std::string& out, u64 arg)
 	});
 }
 
+enum class SearchState
+{
+	not_initialized = 0,
+	idle,
+	in_progress,
+	initializing,
+	canceling,
+	finalizing,
+};
+
 struct search_info
 {
 	vm::ptr<CellSearchSystemCallback> func;
 	vm::ptr<void> userData;
+
+	atomic_t<SearchState> state = SearchState::not_initialized;
+
+	shared_mutex links_mutex;
+	std::unordered_map<std::string, std::string> content_links;
 };
 
+struct search_content_t
+{
+	CellSearchContentType type = CELL_SEARCH_CONTENTTYPE_NONE;
+	CellSearchTimeInfo timeInfo;
+	CellSearchContentInfoPath infoPath;
+	union
+	{
+		CellSearchMusicInfo music;
+		CellSearchPhotoInfo photo;
+		CellSearchVideoInfo video;
+		CellSearchMusicListInfo music_list;
+		CellSearchPhotoListInfo photo_list;
+		CellSearchVideoListInfo video_list;
+		CellSearchVideoSceneInfo scene;
+	} data;
+};
+
+using ContentIdType = std::pair<u64, std::shared_ptr<search_content_t>>;
+using ContentIdMap = std::unordered_map<u64, std::shared_ptr<search_content_t>>;
+
 struct search_object_t
 {
 	// TODO: Figured out the correct values to set here
 	static const u32 id_base  = 1;
 	static const u32 id_step  = 1;
 	static const u32 id_count = 64;
 	static const u32 invalid  = 0xFFFFFFFF;
+
+	std::vector<ContentIdType> content_ids;
 };
 
 error_code cellSearchInitialize(CellSearchMode mode, u32 container, vm::ptr<CellSearchSystemCallback> func, vm::ptr<void> userData)
 {
 	cellSearch.warning("cellSearchInitialize(mode=0x%x, container=0x%x, func=*0x%x, userData=*0x%x)", +mode, container, func, userData);
 
+	if (!func)
+	{
+		return CELL_SEARCH_ERROR_PARAM;
+	}
+
 	const auto search = g_fxo->get<search_info>();
+
+	switch (search->state.load())
+	{
+		case SearchState::not_initialized:
```

Comment:
Afaik our style is that switch and case should be on same indentation level. Same goes for all switch statements in this PR, I'm not going to tag them all.